kube/src/client/config_ext.rs

```diff
@@ -0,0 +1,155 @@
+use std::convert::TryFrom;
+
+use tower::util::Either;
+
+#[cfg(any(feature = "native-tls", feature = "rustls-tls"))] use super::tls;
+use super::{
+    auth::Auth,
+    middleware::{AddAuthorizationLayer, AuthLayer, RefreshTokenLayer, SetBaseUriLayer},
+};
+use crate::{Config, Result};
+
+/// Extensions to [`Config`](crate::Config) for custom [`Client`](crate::Client).
+///
+/// See [`Client::new`](crate::Client::new) for an example.
+///
+/// This trait is sealed and cannot be implemented.
+pub trait ConfigExt: private::Sealed {
```

Comment:
I didn't know about this pattern, but am a fan. Feels like a very smart way to avoid having all the public methods in a big namespace :+1:

Reply:
Yeah, the downside is the extra import, but I think the explicit `kube::client::ConfigExt` actually makes sense for this. I also wanted to keep client stuff out of the config module.